File: AppFilmes/app/src/main/java/com/devheverton/appfilmes/Adapter/AdapterFilme.java
package com.devheverton.appfilmes.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.devheverton.appfilmes.Model.Filme;
import com.devheverton.appfilmes.R;


import java.util.List;

public class AdapterFilme extends RecyclerView.Adapter<AdapterFilme.FilmeViewHolder> {

    private Context context;
    private List<Filme> filmeList;

    public AdapterFilme(Context context, List<Filme> filmeList) {
        this.context = context;
        this.filmeList = filmeList;
    }

    @NonNull
    @Override
    public FilmeViewHolder onCreateViewHolder(@NonNull  ViewGroup parent, int viewType) {
        View itemLista;
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        itemLista = layoutInflater.inflate(R.layout.filmes_item,parent,false);
        return new FilmeViewHolder(itemLista);
    }

    @Override
    public void onBindViewHolder(@NonNull  AdapterFilme.FilmeViewHolder holder, int position) {
        Glide.with(context).load(filmeList.get(position).getCapa()).into(holder.capa);
        holder.titulo.setText(filmeList.get(position).getTitulo());
    }

    @Override
    public int getItemCount() {
        return filmeList.size();
    }

    public class FilmeViewHolder extends RecyclerView.ViewHolder{

        private ImageView capa;
        private TextView titulo;
        private TextView descricao;
        private TextView elenco;

        public FilmeViewHolder(@NonNull View itemView) {
            super(itemView);
            capa = itemView.findViewById(R.id.capaFilme);
            titulo = itemView.findViewById(R.id.tituloFilme);
            descricao = itemView.findViewById(R.id.dt_descricaoFilme);
            elenco = itemView.findViewById(R.id.dt_elencoFilme);
        }
    }
}

File: AppdeFilmes2/app/src/main/java/com/devheverton/appdefilmes/adapter/AdapterFilme.kt
package com.devheverton.appdefilmes.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.devheverton.appdefilmes.databinding.FilmeItemBinding

import com.devheverton.appdefilmes.model.Filme
import com.devheverton.appdefilmes.view.DetalhesFilme

class AdapterFilme (private val context: Context, private val listaFilme: MutableList<Filme>):
    RecyclerView.Adapter<AdapterFilme.FilmeViewHolder>() {



    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilmeViewHolder {
        val itemLista = FilmeItemBinding.inflate(LayoutInflater.from(context),parent,false)
        return FilmeViewHolder(itemLista)
    }


    override fun onBindViewHolder(holder: FilmeViewHolder, position: Int) {
        Glide.with(context).load(listaFilme[position].capa).centerCrop().into(holder.capa)

        holder.capa.setOnClickListener {
            val intent = Intent(context,DetalhesFilme:: class.java)
            context.startActivity(intent)

        }
    }
    override fun getItemCount() = listaFilme.size


    inner class FilmeViewHolder(binding: FilmeItemBinding): RecyclerView.ViewHolder(binding.root){
        val capa = binding.CapaFilme

    }

}